io_scene_vrm/importer/vrm_animation_importer.py

```diff
@@ -0,0 +1,741 @@
+                continue
+            translations = read_accessor_as_animation_sampler_translation_output(
+                output_accessor_dict, buffer_view_dicts, buffer_dicts, buffer0_bytes
+            )
+            if translations is None:
+                continue
+            keyframes = tuple(sorted(zip(timestamps, translations)))
+            node_index_to_translation_keyframes[node_index] = keyframes
+        elif animation_path == "rotation":
+            timestamps = read_accessor_as_animation_sampler_input(
+                input_accessor_dict, buffer_view_dicts, buffer_dicts, buffer0_bytes
+            )
+            if timestamps is None:
+                continue
+            rotations = read_accessor_as_animation_sampler_rotation_output(
+                output_accessor_dict, buffer_view_dicts, buffer_dicts, buffer0_bytes
+            )
+            if rotations is None:
+                continue
+            keyframes = tuple(sorted(zip(timestamps, rotations)))
+            node_index_to_rotation_keyframes[node_index] = keyframes
+
+    timestamps = [
+        timestamp
+        for keyframes in itertools.chain(
+            node_index_to_translation_keyframes.values(),
+            node_index_to_rotation_keyframes.values(),
+        )
+        for (timestamp, _) in keyframes
+    ]
+    timestamps.sort()
+    if not timestamps:
+        return {"CANCELLED"}
+
+    first_timestamp = timestamps[0]
+    last_timestamp = timestamps[-1]
+
+    first_frame_count = math.floor(
+        first_timestamp * context.scene.render.fps / context.scene.render.fps_base
+    )
+    last_frame_count = math.ceil(
+        last_timestamp * context.scene.render.fps / context.scene.render.fps_base
+    )
+    for frame_count in range(first_frame_count, last_frame_count + 1):
+        timestamp = (
+            frame_count * context.scene.render.fps_base / context.scene.render.fps
+        )
+        assign_keyframe(
+            context,
+            armature,
+            action,
+            node_rest_pose_tree,
+            node_index_to_human_bone_name,
+            node_index_to_translation_keyframes,
+            node_index_to_rotation_keyframes,
+            frame_count,
+            timestamp,
+            humanoid_parent_rest_world_matrix=Matrix(),
+            intermediate_rest_local_matrix=Matrix(),
+            intermediate_pose_local_matrix=Matrix(),
+        )
+
+    return {"FINISHED"}
+
+
+def dump(v: Union[Matrix, Vector, Quaternion, float, int]) -> str:
+    if logger.level > logging.DEBUG:
+        return "(omit)"
+
+    if isinstance(v, (float, int)):
+        return str(v)
+
+    if isinstance(v, Matrix):
+        t, r, s = v.decompose()
+        return f"Matrix(T={dump(t)},R={dump(r)},S={dump(s)})"
+
+    if isinstance(v, Vector):
+        return f"({v.x:.3f},{v.y:.3f},{v.z:.3f})"
+
+    x, y, z = [round(math.degrees(xyz)) for xyz in v.to_euler("XYZ")[:]]
+    return f"Euler({x},{y},{z})"
+
+
+def assign_keyframe(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function contains 72 lines of code, not including blank lines or lines with only comments, Python punctuation characters, identifiers, or literals. By comparison, 99% of the functions in the CodeGuru reference dataset contain fewer lines of code. Large functions might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this function or break it into multiple functions.